CMakeLists.txt

```diff
@@ -137,9 +144,14 @@ if(${VELOX_BUILD_TESTING})
   set(VELOX_ENABLE_HIVE_CONNECTOR ON)
   set(VELOX_ENABLE_TPCH_CONNECTOR ON)
   set(VELOX_ENABLE_SPARK_FUNCTIONS ON)
+  set(velox_ENABLE_DWIO ON)
   set(VELOX_ENABLE_EXAMPLES ON)
 endif()
 
+if(${VELOX_BUILD_DWIO})
+  set(VELOX_ENABLE_DWIO ON)
+endif()
```

Comment:
nit: What if we have contradictory build flags - say VELOX_ENABLE_HIVE_CONNECTOR ON, but VELOX_ENABLE_DWIO OFF ,should we check for that ?

Reply:
@kgpai Good question. These flags are a bit all over the place today. I can add a check here, but it seems like we should separate the flags we use internally to control which directories we compile, from the external facing flags that users can toggle, since in many cases they can be incompatible. 

Something like this, for example, won't work as expected today: 

```
$ EXTRA_CMAKE_FLAGS=" -DVELOX_BUILD_MINIMAL=ON -DVELOX_ENABLE_HIVE_CONNECTOR=ON" make
```

Since in line 126 we will overwrite the enable hive connector flag. And even if it does, hive needs exec and the compression libraries, which won't be available. 

Point being that users can't really (or shouldn't) tweak VELOX_ENABLE_HIVE_CONNECTOR manually. 

Cc: @assignUser @majetideepak

Reply:
I added the check in any case since is likely a longer discussion.

Reply:
Yes, reorganizing the options and also giving some nice output at the beginning of the configure phase etc. is on my list but atm lower priority. Adding CMakePresets.json would also be good (partially replaces makefile, get's picked up by IDEs...).